Changed item quantity of a craftable item to 0

ineternet opened the pull request:
Fix for #1604 
Whenever a system has none of a specific item but the ability to craft it, the grid saves the quantity as 1. Quantity sorting would make use of this and sort it alphabetically together with other quantity 1 items. This change should make quantity sorting think craftable items have a quantity of 0, making all of them appear behind itemstacks with the quantity 1 (or higher).

ineternet committed:
Changed item quantity of a craftable item to 0

Whenever a system has none of a specific item but the ability to craft it, the grid saves the quantity as 1. Quantity sorting would make use of this and sort it alphabetically together with other quantity 1 items. This change should make quantity sorting think craftable items have a quantity of 0, making all of them appear behind itemstacks with the quantity 1 (or higher).

raoulvdberge commented:
Can't you change `getQuantity` in `GridStackItem` instead? That should be a little cleaner.

ineternet committed:
Moving code logic to GridStackItem

ineternet committed:
Update GridSortingQuantity.java

ineternet committed:
Changed stack quantity logic

Craftable itemstacks remain stored as "1 item" internally. Other classes will be given the information that the stack is empty, though.

ineternet commented:
`GridStackItem` should now still store craftable items as 1, but tell any other class they're empty. This makes GridSortingQuantity think craftable itemstacks have less items on them than actual single items in the system. Although I'm unsure if any other classes will have a problem with the itemstack being supposedly empty?

raoulvdberge commented:
Shouldn't cause any issues. These are the only usages of the `getQuantity` method:
![x](https://drive.google.com/uc?id=1S0GHXAVSuL2j8GVZmTIB1s1DTWBG3ljv)

raoulvdberge commented:
Can you add a changelog entry again?

ineternet committed:
Update CHANGELOG.md

raoulvdberge commented:
Thank you.
